### DISCUSSION TITLE
Adding image data / buffers to NodeIO writeJSON

### DISCUSSION THREAD

**Discussion by Ellink** [donmccurdy/glTF-Transform#785] [Q&A] [answered]
Hi, 

I'm developing a little nodejs web app with express.js, where i load, optimize and modify a lot of GLTF files, for then to return them to a third-party in json format. 

I can't seem to figure out how to include the image data and buffer in the json output from NodeIO. Is this not possible with this library or have i overlooked anything in the documentation? Any pointers are appreciated :)

**Comment by donmccurdy** [comment, marked as answer]
It should all be in the `resources` entry of the JSON. The output of writeJSON looks like this:

```javascript
const jsonDocument = {
    // glTF JSON schema.
    json: {
        asset: {version: '2.0'},
        images: [{uri: 'image1.png'}, {uri: 'image2.png'}],
        ...
    },

    // URI → Uint8Array mapping.
    resources: {
        'image1.png': new Uint8Array(...),
        'image2.png': new Uint8Array(...),
    }
};
```

https://gltf-transform.donmccurdy.com/interfaces/core.jsondocument.html

**Reply by Ellink** [reply]
Thank you for the quick reply. 
I noticed the resources element, but thought it maybe was possible to add in the images to the GLTF json output as encoded images. I think my solution is going to be to encode buffer and images myself.
